[Pytorch] Add Armv9-a build flags

Nicoshev opened the pull request:
Summary: Adding detection and enablement of Armv9a core instruction sets

Test Plan: CI

Differential Revision: D85860431




cc @snadampal @milpuz01 @aditew01 @nikhil-arm @fadara01 @nWEIdia @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @chenyang78

Nicoshev commented:
@pytorchbot label "release notes: cpu (aarch64)"

malfet commented on torch/_inductor/cpu_vec_isa.py:
```diff
@@ -160,6 +191,22 @@ class VecNEON(VecISA):
+            # restore _arch_flags
+            self._arch_flags = base_flags
+
+        return False
```
To be a bit more precise, IMO you have to call a super method there, that checks if compilation still succeeds with original flags

Nicoshev committed:
[Pytorch] Add Armv9-a build flags (#166640)

Summary:

Adding detection and enablement of Armv9a core instruction sets

Test Plan: CI

Differential Revision: D85860431
